xtask/src/build_docker_image.rs

```diff
@@ -142,25 +187,77 @@ pub fn build_docker_image(
 
         docker_build.args(&["-f", &dockerfile]);
 
-        if std::env::var("GITHUB_ACTIONS").is_ok() || progress == "plain" {
+        if gha || progress == "plain" {
             docker_build.args(&["--progress", "plain"]);
         } else {
             docker_build.args(&["--progress", &progress]);
         }
 
         docker_build.arg(".");
 
-        if force || !push || std::env::var("GITHUB_ACTIONS").is_ok() {
-            docker_build.run(verbose)?;
+        if !dry_run && (force || !push || gha) {
+            let result = docker_build.run(verbose);
+            if gha {
+                if let Err(e) = &result {
+                    // TODO: Determine what instruction errorred, and place warning on that line with appropriate warning
+                    println!("::error file=docker/{dockerfile},title=Build failed::{}", e)
```

Comment:
I'm not sure how valuable this error is 

https://github.com/cross-rs/cross/runs/6881463959?check_suite_focus=true

It would be better if it could capture/tee the stderr/stdout and determine the actual error instead of spitting out what it currently does

Reply:
Won't it cause a failure in the bors action for that target anyway, so we'll know which specific Dockerfile is failing already? I don't think this is necessary.

Reply:
I was hoping that with it, it would show the annotation in the pr, but since it doesn't do that it doesn't make much sense.

There is a reason to keep it however if there are multiple targets, I'll change this